perllib/FixMyStreet/Cobrand/Zurich.pm

```diff
@@ -1321,4 +1289,36 @@ sub singleton_bodies_str { 1 }
 
 sub contact_extra_fields { [ 'abbreviation' ] };
 
+sub db_state_migration {
+    my $rs = FixMyStreet::DB->resultset('State');
+
+    # Create new states needed
+    $rs->create({ label => 'feedback pending', type => 'open', name => 'Rückmeldung ausstehend' });
+    $rs->create({ label => 'wish', type => 'closed', name => 'Wunsch' });
+    $rs->create({ label => 'external', type => 'closed', name => 'Extern' });
+    $rs->create({ label => 'jurisdiction unknown', type => 'closed', name => 'Zuständigkeit unbekannt' });
+    $rs->create({ label => 'not contactable', type => 'closed', name => 'Nicht kontaktierbar' });
+
+    # And update used current ones to have correct name
+    $rs->find({ label => 'in progress' })->update({ name => 'In Bearbeitung' });
+    $rs->find({ label => 'fixed' })->update({ name => 'Beantwortet' });
+
+    # Move reports to correct new state
+    my %state_move = (
+        closed => 'external',
+        investigating => 'wish',
+        'unable to fix' => 'jurisdiction unknown',
+        planned => 'feedback pending',
+        partial => 'not contactable',
+    );
+    foreach (keys %state_move) {
+        FixMyStreet::DB->resultset('Problem')->search({ state => $_ })->update({ state => $state_move{$_} });
+    }
+
+    # Delete unused standard states from the database
+    for ('action scheduled', 'duplicate', 'not responsible', 'internal referral', 'planned', 'investigating', 'unable to fix') {
+        $rs->find({ label => $_ })->delete;
+    }
+}
```

Comment:
Is it possible that `closure_status` could be set to any of the old states?

Reply:
Theoretically, though there aren't any at the moment. If I do this end of a week or something, there shouldn't be any, that's only ones not been dealt with.